hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/SqlAnalyzedJobSuspensionIsForbiddenTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl;
+
+import com.hazelcast.cluster.ClusterState;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.SimpleTestInClusterSupport;
+import com.hazelcast.jet.core.JetTestSupport;
+import com.hazelcast.jet.core.JobStatus;
+import com.hazelcast.jet.impl.JobResult;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Note: We prepared a separate test suite to prevent tests to be flaky
+ *  because of {@link AnalyzeStatementTest} inherit {@link SimpleTestInClusterSupport},
+ *  where cluster members are shared between tests.
+ */
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class SqlAnalyzedJobSuspensionIsForbiddenTest extends JetTestSupport {
+    private HazelcastInstance instance;
+
+    @Before
+    public void setUp() throws Exception {
+        instance = createHazelcastInstance();
+    }
+
+    @Test
+    public void when_suspend_isForbidden() {
+        // When
+        Job job = assertRunQuery();
+
+        // Then
+        assertThatThrownBy(job::suspend)
+                .hasMessageContaining("Cannot suspend the job being analyzed");
+    }
+
+    @Test
+    public void when_restart_isForbidden() {
+        // When
+        Job job = assertRunQuery();
+
+        // Then
+        assertThatThrownBy(job::restart)
+                .hasMessageContaining("Cannot suspend the job being analyzed");
+    }
+
+    @Test
+    public void test_changedClusterState() {
+        // When
+        Job job = assertRunQuery();
+
+        // Then
+        instance.getCluster().changeClusterState(ClusterState.PASSIVE);
+        instance.getCluster().changeClusterState(ClusterState.ACTIVE);
+
+        JobResult jobResult = getJetServiceBackend(instance).getJobRepository().getJobResult(job.getId());
```

Comment:
for `analyze` you should be able to get the job by id and assert state via `Job` instead of using internal objects.

Reply:
BTW, you already have `Job` instance, but just to be sure you might get a fresh one

Reply:
I used that way because it was convenient to me, see no harm here.

Reply:
`instance.getJet().getJob(job.getId()).join();` + `assertThatThrownBy`. There is no direct API to get `getFailureText` from `Job`